### PR TITLE
Pythonic arguments for KaldiRecognizer class for vosk

### PR DESCRIPTION
**Pull request by farooqkz**
Arguments for `__init__` method of the class KaldiRecognizer were correct but not Pythonic. I've made a little change. Please test and merge.

EDIT:

The change is kinda trivial and does not affect the functionality at all.

### CODE REVIEW

File: python/vosk/__init__.py [hunk 1]
```diff
@@ -47,15 +47,15 @@ def __del__(self):
 
 class KaldiRecognizer(object):
 
-    def __init__(self, *args):
-        if len(args) == 2:
+    def __init__(self, stream, sample_rate, model=None):
```
Comment (nshmyrev) [inline python/vosk/__init__.py:50]: What is stream?

File: python/vosk/__init__.py [hunk 2]
```diff
@@ -47,15 +47,15 @@ def __del__(self):
...
+        if model is None:
             self._handle = _c.vosk_recognizer_new(args[0]._handle, args[1])
-        elif len(args) == 3 and type(args[2]) is SpkModel:
-            self._handle = _c.vosk_recognizer_new_spk(args[0]._handle, args[1], args[2]._handle)
-        elif len(args) == 3 and type(args[2]) is str:
+        elif isinstance(model, str):
             self._handle = _c.vosk_recognizer_new_grm(args[0]._handle, args[1], args[2].encode('utf-8'))
```
Comment (nshmyrev) [inline python/vosk/__init__.py]: You removed args above, is it going to work at all?

File: python/vosk/__init__.py (hunk 1 above)
Comment (farooqkz) [reply python/vosk/__init__.py:50]: It's name of one of arguments, I can change it to something else if you want

File: python/vosk/__init__.py (hunk 2 above)
Comment (farooqkz) [reply python/vosk/__init__.py]: Yes. You can test it if you want.

Comment (farooqkz) [reply python/vosk/__init__.py]: hello?

Comment (uuid0) [reply python/vosk/__init__.py]: Hey :)
Wouldn't you want to use the keyword arguments for the sake of clarity?
e.g. `_c.vosk_recognizer_new_grm(args[0]._handle, args[1], args[2].encode('utf-8'))` => `_c.vosk_recognizer_new(stream._handle, sample_rate, model.encode('utf-8'))`

Comment (farooqkz) [reply python/vosk/__init__.py]: that's what my PR does...

Comment (uuid0) [reply python/vosk/__init__.py]: What I meant is:
In the PR the __init__ signature has keyword arguments, but when calling other functions in its body, args is used instead.

Current code in the PR:
```
def __init__(self, stream, sample_rate, model=None):
        if model is None:
            self._handle = _c.vosk_recognizer_new(args[0]._handle, args[1])
        elif isinstance(model, str):
            self._handle = _c.vosk_recognizer_new_grm(args[0]._handle, args[1], args[2].encode('utf-8'))
        elif isinstance(model, SpkModel):
            self._handle = _c.vosk_recognizer_new_spk(args[0]._handle, args[1], args[2]._handle)
        else:
            raise TypeError("model argument must be either None, str or SpkModel")

        if self._handle == _ffi.NULL:
            raise Exception("Failed to create a recognizer")
```

Could be:
```
    def __init__(self, stream, sample_rate, model=None):
        if model is None:
            self._handle = _c.vosk_recognizer_new(stream._handle, sample_rate)
        elif isinstance(model, str):
            self._handle = _c.vosk_recognizer_new_grm(stream._handle, sample_rate, model.encode('utf-8'))
        elif isinstance(model, SpkModel):
            self._handle = _c.vosk_recognizer_new_spk(stream._handle, sample_rate, model._handle)
        else:
            raise TypeError("model argument must be either None, str or SpkModel")

        if self._handle == _ffi.NULL:
            raise Exception("Failed to create a recognizer")
```

Comment (farooqkz) [reply python/vosk/__init__.py]: My bad...
Currently I am tired. So please send a PR to my branch and fork or wait for me to fix it.

**Comment by nshmyrev** [comment]
Sorry, this pull doesn't solve any problems for us and it doesn't follow naming anyway.

**Pull request closed without merging**